### PR TITLE
fix(Button): add white button variant

### PR DESCRIPTION
**Pull request by mtblanton**
Base:
image

Base hover:
image

Base focus-visible:
image

Outline: 
image

Outline hover:
image

# Github Issue or Trello Card
This PR addresses this issue: #597

# What type of change is this?
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updating documentation.
- [ ] Updating deployment/build pipeline.
- [ ] Updating Project Dependencies
- [ ] Improving or adding to Test Coverage

# Completeness Checklist

- [x] TESTS: My changes maintain the baseline required test coverage, as specified by code climate analysis.
- [x] DOCS: All new component work is covered in that component's `Overview` docs.
- [x] DOCS: All new component work is covered in a component `Playground`.
- [x] DOCS: All new visual changes or options are covered under relevant components' `VisualTests`.
- [x] My code has no linting or typescript compile warnings.
- [x] My work is tied to a Github issue and satisfies the acceptance criteria (if applicable) of the corresponding issue.

# Quality Checklist:

- [ ] My code follows the style guidelines of this project - possibly not due to needing a transparent color for the background
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# UI Checklist
- [x] I have conducted visual UAT on my changes/features.
- [x] My solution works well on desktop, tablet, and mobile browsers.

### CODE REVIEW

File: src/components/Button/Button.Overview.stories.mdx [hunk 1]
```diff
@@ -56,6 +57,11 @@ the color of the primary CTA but makes the hierarchy clear. The background is tr
       <Button variant="dark" isOutlined>
         Dark
       </Button>
+      <Box background="primary" padding="2xs">
```
Comment (mtblanton) [inline src/components/Button/Button.Overview.stories.mdx:60]: I added this background so the button is visible, if you all have a better solution for this please LMK!

**Comment by juanfabrega** [comment]
@nathanyoung I think you should probably review this one from a visual standpoint.

**Review by nathanyoung — CHANGES_REQUESTED** [review]
Please add the white variants to our "Visual Regression Tests" stories. 

https://github.com/palmetto/palmetto-components/blob/main/src/components/Button/Button.VisualTests.stories.tsx

This will render the white variant with every prop combination

**Comment by nathanyoung** [comment]
[Button - "white" variant](https://trello.com/c/ibEymAf1/917-button-white-variant)

File: src/styles/variables/buttons.scss [hunk 2]
```diff
@@ -91,4 +91,18 @@
   --button-background-hover-color-dark-outline: var(--color-brand-dark-base);
   --button-font-hover-color-dark-outline: var(--color-brand-white-base);
   --button-focus-box-shadow-dark: 0 0 0 4px var(--color-brand-grey-200);
+
+  /* WHITE */
+  --button-background-color-white: var(--color-brand-white-base);
+  --button-font-color-white: var(--color-brand-dark-base);
+  --button-background-hover-color-white: var(--color-brand-grey-200);
```
Comment (nathanyoung) [inline src/styles/variables/buttons.scss:98]:
```suggestion
  --button-background-hover-color-white: var(--color-brand-grey-100);
```

**Comment by nathanyoung** [comment]
@mtblanton This PR looks good, but we're going to hold on merging this for now until the design team has had a chance to think about what we really want to do here.

**Comment by nathanyoung** [comment]
> 

Let me bring this up with the team in standup tomorrow

**Review by nathanyoung — APPROVED** [review]

**Pull request merged**